### PR TITLE
[gha] allure related tools.

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Allure tooling, in prep for allure report action.
While working on dev-setup.sh, removed centos docker image.
Also clean up script quite a bit.
All tools (aside from lcov which we don't really need we can use grcov today) are install in alpine.
Also coverage tools are likely to change once I fix it.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

## Related PRs

https://github.com/diem/actions/pull/17

## If targeting a release branch, please fill the below out as well

 * Justification and breaking nature (who does it affect? validators, full nodes, tooling, operators, AOS, etc.)
 * Comprehensive test results that demonstrate the fix working and not breaking existing workflows.
 * Why we must have it for V1 launch.
 * What workarounds and alternative we have if we do not push the PR.

### CODE REVIEW

File: .github/actions/docker-checks/action.yml [hunk 1]
```diff
@@ -30,5 +30,4 @@ runs:
       run: |
         hadolint docker/ci/github/Dockerfile && \
         hadolint docker/ci/arch/Dockerfile && \
-        hadolint docker/ci/alpine/Dockerfile && \
-        hadolint docker/ci/centos/Dockerfile
```
Comment (rexhoffman) [inline .github/actions/docker-checks/action.yml:34]: Removing centos as the project is dead so further cleanup makes little sense.

File: .github/workflows/ci-test.yml [hunk 2]
```diff
@@ -128,8 +128,9 @@ jobs:
     needs: prepare
     if: ${{ needs.prepare.outputs.test-dev-setup == 'true' }}
     strategy:
+      fail-fast: false
```
Comment (rexhoffman) [inline .github/workflows/ci-test.yml:131]: this will let all docker builds run to completion.

File: docker/ci/alpine/Dockerfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ COPY scripts/dev_setup.sh /diem/scripts/dev_setup.sh
 RUN apk add bash=5.1.0-r0 --no-cache
 
 # Batch mode and all operations tooling
-RUN scripts/dev_setup.sh -t -o -y -b -p
+RUN scripts/dev_setup.sh -t -o -y -b -p -s
```
Comment (rexhoffman) [inline docker/ci/alpine/Dockerfile:11]: test installing all codegen tools as well in alpine

File: docker/ci/alpine/Dockerfile [hunk 4]
```diff
@@ -27,10 +27,19 @@ RUN [ -x "$(command -v shellcheck)" ] \
     && [ -x "$(command -v helm)" ] \
     && [ -x "$(command -v aws)" ] \
     && [ -x "$(command -v z3)" ] \
+    && [ -x "$(command -v tidy)" ] \
```
Comment (rexhoffman) [inline docker/ci/alpine/Dockerfile:30]: add the tools I can about and the ones that should have been add for other language support

File: scripts/dev_setup.sh [hunk 5]
```diff
@@ -11,6 +11,15 @@
 # 2 .profile will be used to configure the shell
 # 3 ${HOME}/bin/ is expected to be on the path - hashicorp tools/hadolint/etc.  will be installed there on linux systems.
 
+# fast fail.
+set -eo pipefail
```
Comment (rexhoffman) [inline scripts/dev_setup.sh:15]: fast fail all errors in the script.   it's far more touchy now and all commands must return the success code.   mostly lead to ensuring grep didn't return error codes....

File: scripts/dev_setup.sh [hunk 6]
```diff
@@ -24,6 +33,8 @@ Z3_VERSION=4.8.9
 CVC4_VERSION=aac53f51
 DOTNET_VERSION=3.1
 BOOGIE_VERSION=2.8.32
+PYRE_CHECK_VERSION=0.0.59
```
Comment (rexhoffman) [inline scripts/dev_setup.sh:36]: these should have had versions added in when they were created.

File: scripts/dev_setup.sh [hunk 7]
```diff
@@ -95,11 +100,17 @@ function install_build_essentials {
...
   # Install Rust
-  [[ "${BATCH_MODE}" == "false" ]] && echo "Installing Rust......"
-  if rustup --version &>/dev/null; then
-	   [[ "${BATCH_MODE}" == "false" ]] && echo "Rust is already installed"
+  if [[ "${BATCH_MODE}" == "false" ]]; then
+     echo "Installing Rust......"
+  fi
+  VERSION="$(rustup --version || true)"
+  if [ -n "$VERSION" ]; then
+	   if [[ "${BATCH_MODE}" == "false" ]]; then
```
Comment (sausagee) [inline scripts/dev_setup.sh:111]: nit - indentation alignment

File: scripts/dev_setup.sh [hunk 8]
```diff
@@ -95,11 +100,17 @@ function install_build_essentials {
...
+  echo installing rust.
   BATCH_MODE=$1
   # Install Rust
-  [[ "${BATCH_MODE}" == "false" ]] && echo "Installing Rust......"
-  if rustup --version &>/dev/null; then
-	   [[ "${BATCH_MODE}" == "false" ]] && echo "Rust is already installed"
+  if [[ "${BATCH_MODE}" == "false" ]]; then
+     echo "Installing Rust......"
+  fi
+  VERSION="$(rustup --version || true)"
```
Comment (sausagee) [inline scripts/dev_setup.sh:109]: Not sure if i see the need for or-ing here.  If rustup is not installed, the env var will be empty and satisfies the check below.

File: scripts/dev_setup.sh [hunk 9]
```diff
@@ -286,6 +303,44 @@ function install_openssl_dev {
...
+  fi
+  if [[ "$PACKAGE_MANAGER" == "apt-get" ]]; then
+    install_pkg lcov "$PACKAGE_MANAGER"
+  fi
+  if [[ "$PACKAGE_MANAGER" == "yum" ]] || [[ "$PACKAGE_MANAGER" == "dnf" ]]; then
+    install_pkg lcov "$PACKAGE_MANAGER"
+  fi
+  if [[ "$PACKAGE_MANAGER" == "pacman" ]] || [[ "$PACKAGE_MANAGER" == "brew" ]]; then
+    echo nope no lcov for you.
+    # install_pkg git "$PACKAGE_MANAGER"
```
Comment (sausagee) [inline scripts/dev_setup.sh:320]: Probably more useful to echo this block than as comment here.

**Review by sausagee — APPROVED** [review]
There are a few formatting nits.  Looks good in all.

File: scripts/dev_setup.sh (hunk 8 above)
Comment (rexhoffman) [reply scripts/dev_setup.sh:109]: sub shell will fail if rustup isn't there.   resulting in an error.

File: scripts/dev_setup.sh (hunk 9 above)
Comment (rexhoffman) [reply scripts/dev_setup.sh:320]: hmm good idea.

**Review by sausagee — DISMISSED** [review]

**Pull request closed without merging**